src/components/WwwFrame/Header/KvAtbModalContainer.vue

```diff
@@ -252,15 +253,31 @@ const fetchPostCheckoutAchievements = async loanIds => {
 			oneAwayText.value = `${target - 1} of ${target}`;
 			showModalContent.value = true;
 			modalVisible.value = true;
-		} else if (addedLoan.value?.basketSize < BASKET_LIMIT_SIZE_FOR_EXP) {
-			showModalContent.value = contributingAchievements.value.length;
+		}
+
+		// eslint-disable-next-line max-len
+		if (addedLoan.value?.basketSize < BASKET_LIMIT_SIZE_FOR_EXP || contributingAchievements.value.length !== achievementsFromBasket.value.length) {
+			achievementsFromBasket.value = [...contributingAchievements.value];
+			showModalContent.value = true;
 			modalVisible.value = true;
 		}
 	}).catch(e => {
 		logFormatter(e, 'Modal ATB Post Checkout Achievements Query');
 	});
 };
 
+const fetchAchievementFromBasket = async () => {
+	await apollo.query({
+		query: postCheckoutAchievementsQuery,
+		variables: { loanIds: loansIdsInBasket.value },
+	}).then(({ data }) => {
+		const loanAchievements = data.postCheckoutAchievements?.overallProgress ?? [];
+		achievementsFromBasket.value = loanAchievements.filter(achievement => achievement.postCheckoutTier !== achievement.preCheckoutTier); // eslint-disable-line max-len
+	}).catch(e => {
+		logFormatter(e, 'Modal ATB Post Checkout Achievements Query');
```

Comment:
We should make this message more unique, since we now have the same message in two locations in this file, which will make it tougher to debug.